src/net/stream.rs

```diff
@@ -0,0 +1,157 @@
+        let ((local, wait), dst) = World::current(|world| {
+            let dst = world.lookup(addr);
+            (world.connect(dst), dst)
+        });
+
+        let peer = wait
+            .await
+            .map_err(|_| io::Error::new(io::ErrorKind::ConnectionRefused, dst.to_string()))?;
+
+        let pair = SocketPair { local, peer };
+        let notify = World::current(|world| world.current_host_mut().finish_connect(pair));
+
+        Ok(Stream::new(pair, notify))
+    }
+
+    fn poll_read_priv(
+        &mut self,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        let read_fut = match self.read_fut.as_mut() {
+            Some(fut) => fut,
+            None => {
+                // fast path if we can recv immediately
+                let maybe = Self::recv(self.pair, buf);
+                if let Some(res) = maybe {
+                    return Poll::Ready(res);
+                }
+
+                let notify = Arc::clone(&self.notify);
+                self.read_fut
+                    .get_or_insert(ReusableBoxFuture::new(
+                        async move { notify.notified().await },
+                    ))
+            }
+        };
+
+        match read_fut.poll(cx) {
+            Poll::Ready(_) => {
+                // Loop until we recv a segment from the host or the read future
+                // is pending. This is necessary as we might be notified, but
+                // the segment is still "on the network" and we need to continue
+                // polling.
+                loop {
+                    let notify = Arc::clone(&self.notify);
+                    read_fut.set(async move { notify.notified().await });
+
+                    match Self::recv(self.pair, buf) {
+                        Some(res) => return Poll::Ready(res),
+                        _ => {
+                            if let Poll::Pending = read_fut.poll(cx) {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                Poll::Pending
+            }
+            Poll::Pending => Poll::Pending,
+        }
+    }
+
+    fn recv(pair: SocketPair, buf: &mut ReadBuf<'_>) -> Option<io::Result<()>> {
+        match World::current(|world| world.recv_on(pair)) {
+            Some(seg) => match seg {
+                Segment::Data(bytes) => {
+                    buf.put_slice(bytes.as_ref());
+                    return Some(Ok(()));
+                }
+            },
+            _ => None,
+        }
+    }
+
+    fn poll_write_priv(&self, buf: &[u8]) -> Poll<Result<usize, io::Error>> {
+        World::current(|world| {
+            let bytes = Bytes::copy_from_slice(buf);
+            world.embark_on(self.pair, Segment::Data(bytes))
+        });
+
+        Poll::Ready(Ok(buf.len()))
+    }
+}
+
+impl Drop for Stream {
+    fn drop(&mut self) {}
+}
```

Comment:
I assume this is here to add Rst? Might wanna add a TODO?

Reply:
I'll just delete this for now.